test/eip20/safeEip20.js

```diff
@@ -0,0 +1,239 @@
+const { assertRevert } = require('../helpers/assertRevert');
+
+const SafeEIP20Abstraction = artifacts.require('SafeEIP20');
+let HST;
+
+contract('SafeEIP20', (accounts) => {
+  beforeEach(async () => {
+    HST = await SafeEIP20Abstraction.new(10000, 'Simon Bucks', 1, 'SBX', { from: accounts[0] });
+  });
+  
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  it('transfers: should fail when the recipient is the zero address', async () => {
+    await assertRevert(HST.transfer.call(ZERO_ADDRESS, 100, { from: accounts[0] }));
+  });
+  
+  it('transfers-from: should fail when the recipient is the zero address.', async () => {
+    await HST.approve(accounts[1], 100, { from: accounts[0] });
+    const allowance = await HST.allowance.call(accounts[0], accounts[1]);
+    assert.strictEqual(allowance.toNumber(), 100);
+
+    await assertRevert(HST.transferFrom.call(accounts[0], ZERO_ADDRESS, 10, { from: accounts[1] }));
+  });   
+  
+  it('transfers: should fail when the recipient is the contract address', async () => {
+    await assertRevert(HST.transfer.call(HST.address, 100, { from: accounts[0] }));
+  });
+  
+  it('transfers-from: should fail when the recipient is the contract address.', async () => {
+    await HST.approve(accounts[1], 100, { from: accounts[0] });
+    const allowance = await HST.allowance.call(accounts[0], accounts[1]);
+    assert.strictEqual(allowance.toNumber(), 100);
+
+    await assertRevert(HST.transferFrom.call(accounts[0], HST.address, 10, { from: accounts[1] }));
+  });   
+  
+  
+  //Starting from this line, the following tests are the same tests applied to EIP20.
```

Comment:
Is there a way that these tests could be exported from the other test file, and run on the safeEIP20 implementation? I'd prefer not to have to copy/paste code changes made from one to the next.

Reply:
Upon your comment, I Encapsulated the EIP compatible functions' tests inside a function and make it called for both EIP20 and SafeEIP20 smart contracts. And the duplicated tests code at safeEip20.js has been deleted.
So now, there is no copy/past code.